### ISSUE TITLE
Fails to build on OS X 10.14.6

### ISSUE THREAD

**Issue by buzmeg** [rust-skia/rust-skia#228]
```
$ RUST_BACKTRACE=full cargo build --release
   Compiling skia-bindings v0.21.0 (/Users/andrewl/rust/rust-skia/skia-bindings)
error: failed to run custom build command for `skia-bindings v0.21.0 (/Users/andrewl/rust/rust-skia/skia-bindings)`

Caused by:
  process didn't exit successfully: `/Users/andrewl/rust/rust-skia/target/release/build/skia-bindings-cd8c0230db54cda3/build-script-build` (exit code: 101)
--- stdout
HOST: x86_64-apple-darwin
STARTING A FULL BUILD
Probing 'python'
Python 2 found: "python"
Skipping "../src".
skia/buildtools                      @ 505de88083136eefd056e5ee4ca0f01fe9b33de8
skia/common                          @ 9737551d7a52c3db3262db5856e6bcd62c462b92
skia/third_party/externals/dawn      @ fd90d767ba58faedc1fa74363ebcb2eb99159ba1
skia/third_party/externals/expat     @ e5aa0a2cb0a5f759ef31c0819dc67d9b14246a4a
skia/third_party/externals/dng_sdk   @ c8d0c9b1d16bfda56f15165d39e0ffa360a11123
skia/third_party/externals/imgui     @ d38d7c6628bebd02692cfdd6fa76b4d992a35b75
skia/third_party/externals/libpng    @ 386707c6d19b974ca2e3db7f5c61873813c6fe44
skia/third_party/externals/angle2    @ ed6e230ec65e7fe4acff17a00eb85bc3c8459bb5
skia/third_party/externals/libjpe... @ 574f3a772c96dc9db2c98ef24706feb3f6dbda9a
skia/third_party/externals/lua       @ e354c6355e7f48e087678ec49e340ca0696725b1
skia/third_party/externals/opencl... @ 4e6d30e406d2e5a65e1d65e404fe6df5f772a32b
skia/third_party/externals/opengl... @ 14b80ebeab022b2c78f84a573f01028c96075553
skia/third_party/externals/opencl... @ 932ed55c85f887041291cef8019e54280c033c35
skia/third_party/externals/microh... @ 748945ec6f1c67b7efc934ab0808e1d32f2fb98d
skia/third_party/externals/freetype  @ 05439f5cc69eaa3deaf3db52a7999af09a2c293a
skia/third_party/externals/harfbuzz  @ 1bada656a86e9cb27d4a6b9fcc50748f0bd9c1d9
skia/third_party/externals/spirv-... @ 0c4feb643b89d1792b02f7cbef315e9d95633bd7
skia/third_party/externals/egl-re... @ a0bca08de07c7d7651047bedc0b653cfaaa4f2ae
skia/third_party/externals/zlib      @ 47af7c547f8551bd25424e56354a2ae1e9062859
skia/third_party/externals/libwebp   @ 0fe1a89dbf1930fc2554dbe76adad5d962054ead
skia/third_party/externals/icu       @ 407b39301e71006b68bd38e770f35d32398a7b14
skia/third_party/externals/piex      @ bb217acdca1cc0c16b704669dd6f91a1b509c406
skia/third_party/externals/spirv-... @ 29c11140baaf9f7fdaa39a583672c556bf1795a1
skia/third_party/externals/spirv-... @ 53ab2144b90abede33be5161aec5dfc94ddc3caf
skia/third_party/externals/wuffs     @ 4080840928c0b05a80cda0d14ac2e2615f679f1a
skia/third_party/externals/sdl       @ 5d7cfcca344034aff9327f77fc181ae3754e7a90
skia/third_party/externals/sfntly    @ b55ff303ea2f9e26702b514cf6a3196a2e3e2974
skia/third_party/externals/swifts... @ 6652f0b6428777b5a4a3d191cc30d8b31366b999
Skia args: is_official_build=true skia_use_system_libjpeg_turbo=false skia_use_system_libpng=false skia_use_libwebp=false skia_use_system_zlib=false skia_enable_skottie=true skia_use_xps=false skia_use_dng_sdk=false skia_enable_particles=true cc="clang" cxx="clang++" skia_use_icu=false skia_use_expat=false extra_cflags=["-DSKIA_IMPLEMENTATION=1"]
Done. Made 47 targets from 31 files in 73ms
ninja: Entering directory `/Users/andrewl/rust/rust-skia/target/release/build/skia-bindings-d108fcc208cf2ff7/out/skia'
ninja: no work to do.
cargo:rerun-if-changed=src/bindings.cpp
cargo:rerun-if-changed=/Users/andrewl/rust/rust-skia/skia-bindings/skia/include
COMPILING BINDINGS: ["src/bindings.cpp"]
TARGET = Some("x86_64-apple-darwin")
OPT_LEVEL = Some("3")
HOST = Some("x86_64-apple-darwin")
CXX_x86_64-apple-darwin = None
CXX_x86_64_apple_darwin = None
HOST_CXX = None
CXX = None
CXXFLAGS_x86_64-apple-darwin = None
CXXFLAGS_x86_64_apple_darwin = None
HOST_CXXFLAGS = None
CXXFLAGS = None
CRATE_CC_NO_DEFAULTS = None
DEBUG = Some("false")
CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2,sse3,ssse3")
running: "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/Users/andrewl/rust/rust-skia/skia-bindings/skia" "-Wall" "-Wextra" "-std=c++14" "-DNDEBUG=" "-DSK_ASSUME_GL=1" "-DSK_GAMMA_APPLY_TO_A8=" "-DSKIA_IMPLEMENTATION=1" "-DSK_GL=" "-DSK_SUPPORT_PDF=" "-DSK_HAS_JPEG_LIBRARY=" "-DSK_HAS_PNG_LIBRARY=" "-o" "/Users/andrewl/rust/rust-skia/target/release/build/skia-bindings-d108fcc208cf2ff7/out/skia/src/bindings.o" "-c" "src/bindings.cpp"
exit code: 0
AR_x86_64-apple-darwin = None
AR_x86_64_apple_darwin = None
HOST_AR = None
AR = None
running: "ar" "crs" "/Users/andrewl/rust/rust-skia/target/release/build/skia-bindings-d108fcc208cf2ff7/out/skia/libskia-bindings.a" "/Users/andrewl/rust/rust-skia/target/release/build/skia-bindings-d108fcc208cf2ff7/out/skia/src/bindings.o"
exit code: 0
cargo:rustc-link-lib=static=skia-bindings
cargo:rustc-link-search=native=/Users/andrewl/rust/rust-skia/target/release/build/skia-bindings-d108fcc208cf2ff7/out/skia
CXXSTDLIB_x86_64-apple-darwin = None
CXXSTDLIB_x86_64_apple_darwin = None
HOST_CXXSTDLIB = None
CXXSTDLIB = None
cargo:rustc-link-lib=c++
GENERATING BINDINGS
cargo:warning=couldn't execute `llvm-config --prefix` (error: No such file or directory (os error 2))
cargo:warning=set the LLVM_CONFIG_PATH environment variable to a valid `llvm-config` executable

--- stderr
clang version 9.0.0 (tags/RELEASE_900/final)
Target: x86_64-apple-darwin
Thread model: posix
InstalledDir: 
ignoring nonexistent directory "../include/c++/v1"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "lib/clang/9.0.0/include"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /Users/andrewl/rust/rust-skia/skia-bindings/skia
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1
 /usr/local/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /usr/include
 /System/Library/Frameworks
 /Library/Frameworks
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/include/xmmintrin.h:1934:3: error: use of undeclared identifier '__builtin_ia32_storehps'
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/include/xmmintrin.h:1951:3: error: use of undeclared identifier '__builtin_ia32_storelps'
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/include/xmmintrin.h:1934:3: error: use of undeclared identifier '__builtin_ia32_storehps', err: true
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/include/xmmintrin.h:1951:3: error: use of undeclared identifier '__builtin_ia32_storelps', err: true
thread 'main' panicked at 'Unable to generate bindings: ()', src/libcore/result.rs:1084:5
stack backtrace:
   0:        0x10422e842 - std::panicking::default_hook::{{closure}}::h3f994bbc901f9889
   1:        0x10422e50d - std::panicking::default_hook::h6c261b7dad1af707
   2:        0x10422ef80 - std::panicking::rust_panic_with_hook::hd3c20890ac648923
   3:        0x10422eaed - std::panicking::continue_panic_fmt::hf444d349a369432b
   4:        0x10422ea49 - rust_begin_unwind
   5:        0x104262a02 - core::panicking::panic_fmt::h7bf9e431fd27d405
   6:        0x104262ae9 - core::result::unwrap_failed::hcd64fedd568f6623
   7:        0x103ff9149 - build_script_build::build_support::skia::build::h2a246987cb2c6034
   8:        0x103fecf07 - build_script_build::main::h0674ddf8a9e3a5d5
   9:        0x103fdefa6 - std::rt::lang_start::{{closure}}::h42e12a259fa91b49
  10:        0x10422e9c8 - std::panicking::try::do_call::h8ffcb2a43dbb732c
  11:        0x1042373cf - __rust_maybe_catch_panic
  12:        0x10422f3fe - std::rt::lang_start_internal::ha688fd553625ef6f
  13:        0x103fed489 - main
```

**Comment by pragmatrix** [comment]
This problem is probably caused by a side-by-side installation of Apple's LLVM 11 (XCode) and LLVM 9 (homebrew?). Somehow bindgen picks up the LLVM 9 libclang instead of Apple's, like it does a few steps before for compiling the bindings.

I'm trying to reproduce this to see if there is a way to detect and handle this situation directly in the skia-bindings build scripts. For the meantime, uninstalling or reinstalling LLVM 9 _may_ put everything back into order.

Perhaps related: https://github.com/rust-skia/rust-skia/pull/114#issuecomment-492939987

**Comment by buzmeg** [comment]
Wiping out the homebrew llvm helped quite a bit:

```
$ cargo build --release
    Updating crates.io index
    Updating git repository `https://github.com/servo/rust-offscreen-rendering-context`
  Downloaded serde v1.0.103
  Downloaded serde_json v1.0.44
  Downloaded libc v0.2.66
  Downloaded unicode-width v0.1.7
  Downloaded cc v1.0.48
   Compiling memchr v2.2.1
   Compiling cc v1.0.48
   Compiling version_check v0.1.5
   Compiling libc v0.2.66
   Compiling pkg-config v0.3.17
   Compiling bitflags v1.2.1
   Compiling glob v0.3.0
   Compiling byteorder v1.3.2
   Compiling log v0.4.8
   Compiling lazy_static v1.4.0
   Compiling proc-macro2 v1.0.6
   Compiling cfg-if v0.1.10
   Compiling serde v1.0.103
   Compiling unicode-xid v0.2.0
   Compiling regex-syntax v0.6.12
   Compiling unicode-width v0.1.7
   Compiling crc32fast v1.2.0
   Compiling quick-error v1.2.2
   Compiling ryu v1.0.2
   Compiling strsim v0.8.0
   Compiling vec_map v0.8.1
   Compiling adler32 v1.0.4
   Compiling bindgen v0.52.0
   Compiling ansi_term v0.11.0
   Compiling curl v0.4.25
   Compiling termcolor v1.0.5
   Compiling lazycell v1.2.1
   Compiling peeking_take_while v0.1.2
   Compiling itoa v0.4.4
   Compiling unicode-segmentation v1.6.0
   Compiling shlex v0.1.1
   Compiling nom v4.2.3
   Compiling thread_local v0.3.6
   Compiling clang-sys v0.28.1
   Compiling textwrap v0.11.0
   Compiling humantime v1.3.0
   Compiling miniz_oxide v0.3.5
   Compiling heck v0.3.1
   Compiling libloading v0.5.2
   Compiling libz-sys v1.0.25
   Compiling curl-sys v0.4.24
   Compiling atty v0.2.13
   Compiling filetime v0.2.8
   Compiling xattr v0.2.2
   Compiling which v3.1.0
   Compiling socket2 v0.3.11
   Compiling aho-corasick v0.7.6
   Compiling rustc-hash v1.0.1
   Compiling quote v1.0.2
   Compiling flate2 v1.0.13
   Compiling clap v2.33.0
   Compiling tar v0.4.26
   Compiling regex v1.3.1
   Compiling cexpr v0.3.6
   Compiling serde_json v1.0.44
   Compiling toml v0.5.5
   Compiling env_logger v0.7.1
   Compiling skia-bindings v0.21.0 (/Users/andrewl/rust/rust-skia/skia-bindings)
warning: couldn't execute `llvm-config --prefix` (error: No such file or directory (os error 2))
warning: set the LLVM_CONFIG_PATH environment variable to a valid `llvm-config` executable
   Compiling skia-safe v0.21.0 (/Users/andrewl/rust/rust-skia/skia-safe)
    Finished release [optimized] target(s) in 12m 52s

```

**Comment by pragmatrix** [comment]
Note that it should be possible to build with LLVM 9 by setting the following environment variables: 

```
export PATH="/usr/local/opt/llvm/bin:$PATH"
export CPPFLAGS="-I/usr/local/opt/llvm/include"
export LDFLAGS="-L/usr/local/opt/llvm/lib"
```

as propted after a `brew install llvm`.

The mixup happens because the package [clang-sys](https://crates.io/crates/clang-sys) locates `libclang.dylib` in both directories, the one from LLVM 9 and the other one from Apple LLVM (XCode), but then decides (wrongly?) to use LLVM 9 even though `clang` in the PATH is Apple's. 

For example, here are the directories in which clang-sys locates `libclang.dylib` on my machine:

```
[clang-sys 0.28.1] DIR: ("/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib", "libclang.dylib", [])
[clang-sys 0.28.1] DIR: ("/Library/Developer/CommandLineTools/usr/lib", "libclang.dylib", [])
[clang-sys 0.28.1] DIR: ("/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib", "libclang.dylib", [])
[clang-sys 0.28.1] DIR: ("/usr/local/opt/llvm/lib", "libclang.dylib", [])
[clang-sys 0.28.1] DIR: ("/usr/local/opt/llvm@9/lib", "libclang.dylib", [])
```

And without version information in the filename, clang-sys decides for the latest one it finds.

I don't see a simple way how I can detect this or prevent it from happening. Also I don't think that this behavior is an error in clang-sys, because it can only _assume_ the libclang location if there is no llvm-config tool available. The only thing I can do, is to clearly state in the readme that this is most likely a configuration problem (included in #237). 

Closed as wontfix.
